qp_qiime2/__init__.py

```diff
@@ -38,21 +39,28 @@
 
 # PLEASE READ:
 # There are 2 main steps:
-# 1. We are going loop over QIITA_Q2_SEMANTIC_TYPE (lookup table) so we can
+# 1. We are going to loop on the Q2_EXTRA_COMMANDS to check which extra
+# commands we want to add, extra from those given in the next step
+# 2. We are going loop over QIITA_Q2_SEMANTIC_TYPE (lookup table) so we can
 # retrieve the q2plugin and their methods that work with that given Q2/Qiita
 # semantic type. Then we will ignore any plugin not in Q2_ALLOWED_PLUGINS so
 # we avoid adding plugins that we don't want; like deblur or dada2. Finally,
 # we are going to loop over the different inputs, outputs and parameters from
 # Q2 and convert them to QIITA's req_params, opt_params and outputs.
-# 2. We are going to "force" add commands that are not added in the previous
-# loop based on user demand
 #
 # Note that Qiita users like to have descriptions of the paramters
 # (q2-description) vs. the parameter itself (q2-parameter) so to allow this
 # we are going to store each parameter twice: one in the
 # opt_params[q2-description]: value; and
 # req_params['qp-hide-param' + q2-description]: q2-parameter
 
+pm = PluginManager()
```

Comment:
This block, up to line 62, is the replacement of all the code below 278 to add `methods['classify_sklearn']`. I guess this highlights the spirit of the changes: realized that much of the code was duplicated to add that method and that we could simplify it if we first just create a list of the methods we will add (`methods_to_add`) and then add them.